extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeThrottlingDecrypter.java

```diff
@@ -66,7 +66,19 @@ public YoutubeThrottlingDecrypter() throws ParsingException {
 
     private String parseDecodeFunctionName(final String playerJsCode)
             throws Parser.RegexException {
-        return Parser.matchGroup1(FUNCTION_NAME_PATTERN, playerJsCode);
+            String functionName = Parser.matchGroup1(FUNCTION_NAME_PATTERN, playerJsCode);
+            int arrayStartBrace = functionName.indexOf("[");
+            
+            if (arrayStartBrace > 0) {
```

Comment:
Are both paths of the if case covered by tests? Because I don't quite understand why this is needed and cannot be adjusted in the regex

Reply:
The condition is always true for the tests we have currently

Reply:
I tested a few more video ids and the condition was still true. However, I do not see a reason to remove the condition. I'd keep it as it is.

Reply:
The branching doesn't depend on video, but on the player version. To test the other case, you have to force an older player like https://www.youtube.com/s/player/8040e515/player_ias.vflset/en_US/base.js